internal/ready/ready.go

```diff
@@ -150,10 +159,37 @@ func (r *ReadinessChecker) unstructuredConditionStatus(obj *unstructured.Unstruc
 			}
 		}
 	}
+
 	// This is a legitimate "unknown" case because we didn't see the condition
 	return "", corev1.ConditionUnknown, nil
 }
 
+// statusField inspects a single top level field on the status
+func (r *ReadinessChecker) statusField(obj *unstructured.Unstructured, conditionType string) (string, corev1.ConditionStatus, error) {
+	// In this case the condition type is "redskyops.dev/status-<FIELD>-<VALUE>" so we must parse out the field and value
+	kv := strings.SplitN(strings.TrimPrefix(conditionType, ConditionTypeStatus), "-", 2)
+
+	s, ok := obj.UnstructuredContent()["status"].(map[string]interface{})
+	if !ok {
+		return "", corev1.ConditionFalse, fmt.Errorf("unable to locate status")
+	}
+
+	v, ok := s[kv[0]].(string)
+	if !ok {
+		// This is a legitimate "unknown" case because we didn't see the field
+		return "", corev1.ConditionUnknown, nil
+	}
+
+	// Check the value case-insensitively
+	if len(kv) > 1 && strings.EqualFold(kv[1], v) {
```

Comment:
I wasn't sure if `len(kv) != 2` should just be an error or not.

Reply:
Yeah, I think it'd be worth adding some additional validation on this to ensure we have a `status-field-value` pattern. Probably after the split, `if len(kv) != 2` then fail/error.